rest/models/snapshot.go

```diff
@@ -69,6 +69,104 @@ func (p GetGainersLosersSnapshotParams) WithIncludeOTC(q bool) *GetGainersLosers
 	return &p
 }
 
+// ListOptionsChainParams is a set of parameters for the ListOptionsChainSnapshot method.
+type ListOptionsChainParams struct {
+	// The underlying ticker symbol of the option contract.
+	UnderlyingAsset string `validate:"required" path:"underlyingAsset"`
+
+	// The strike price of the option contract.
+	StrikePrice *float64 `query:"string_price"`
+
+	// The type of contract. Can be ContractCall, ContractPut, or in some rare cases, ContractOther.
+	ContractType *ContractType `query:"contract_type"`
+
+	// The contract's expiration date in YYYY-MM-DD format.
+	ExpirationDateEQ  *Date `query:"expiration_date"`
+	ExpirationDateLT  *Date `query:"expiration_date.lt"`
+	ExpirationDateLTE *Date `query:"expiration_date.lte"`
+	ExpirationDateGT  *Date `query:"expiration_date.gt"`
+	ExpirationDateGTE *Date `query:"expiration_date.gte"`
+
+	// Limit the number of results returned, default is 10 and max is 1000.
+	Limit *int `query:"limit"`
+
+	// Sort field used for ordering.
+	Sort *Sort `query:"sort"`
+
+	// Order results based on the sort field.
+	Order *Order `query:"order"`
+}
+
+// WithUnderlyingAsset the underlying ticker symbol of the option contract.
+func (o ListOptionsChainParams) WithUnderlyingAsset(ticker string) *ListOptionsChainParams {
```

Comment:
We generally only provide these helpers for query params since they're pointers and a bit annoying to use without helpers. The usage pattern is something like this:
```golang
res, err := c.GetDailyOpenCloseAgg(context.Background(), models.GetDailyOpenCloseAggParams{
	Ticker: "AAPL",
	Date:   models.Date(time.Date(2020, 10, 14, 0, 0, 0, 0, time.Local)),
}.WithAdjusted(true))
```

Reply:
makes sense. removing now.